packages/react-instantsearch-dom/src/components/__tests__/VoiceSearch.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import Enzyme, { mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import VoiceSearch from '../VoiceSearch';
+
+const mockGetState = jest.fn().mockImplementation(() => ({}));
+const mockIsBrowserSupported = jest.fn().mockImplementation(() => true);
+const mockIsListening = jest.fn();
+const mockToggleListening = jest.fn();
```

Comment:
`../../lib/voiceSearchHelper` exports a function `voiceSearchHelper` and `voiceSearchHelper()` returns an object containing `getState, isBrowserSupported, isListening, toggleListening`.
Do you know how I can achieve this with inlining them?

```js
      const wrapper = mount(<VoiceSearch />);
      wrapper.find('button').simulate('click');
      expect(mockToggleListening).toHaveBeenCalledTimes(1);
```

Apparently the instance of `voiceSearchHelper` is not visible here. So I don't know how to get the mocked version of `toggleListening` here with inlining way.

Reply:
You're correct it's not visible. We could access it through the component instance but it's a terrible idea. A solution would have been to decoupled a bit the `VoiceSearch` component from the `VoiceSearchHelper`. We could have create the service outside of the component (at the widget level for example) to pass it as prop. It would have been a lot easier to test. Anyway no problem to keep it as is.